Makefile.common

```diff
@@ -0,0 +1,300 @@
+# Copyright 2018 The Prometheus Authors
```

Comment:
We're not using any of this yet, but I guess it doesn't hurt to have it around and start using some of these common rules in the future.